cmd/gitops/beta/run/cmd.go

```diff
@@ -48,8 +72,268 @@ func betaRunCommandPreRunE(endpoint *string) func(*cobra.Command, []string) erro
+	if err != nil {
+		return fmt.Errorf("install failed: %w", err)
+	}
+
+	fmt.Println(applyOutput)
+
+	return nil
+}
+
+type rootFlags struct {
+	// timeout      time.Duration
+	// verbose      bool
+	pollInterval time.Duration
+	defaults     install.Options
+}
+
+func newManager(kubeClient ctrlclient.Client, rcg genericclioptions.RESTClientGetter) (*ssa.ResourceManager, error) {
+	restMapper, err := rcg.ToRESTMapper()
+	if err != nil {
+		return nil, err
+	}
+
+	kubePoller := polling.NewStatusPoller(kubeClient, restMapper, polling.Options{})
+
+	return ssa.NewResourceManager(kubeClient, kubePoller, ssa.Owner{
+		Field: "flux",
+		Group: "fluxcd.io",
+	}), nil
+}
+
+func applySet(kubeClient ctrlclient.Client, ctx context.Context, rcg genericclioptions.RESTClientGetter, objects []*unstructured.Unstructured) (*ssa.ChangeSet, error) {
+	man, err := newManager(kubeClient, rcg)
+	if err != nil {
+		return nil, err
+	}
+
+	return man.ApplyAll(ctx, objects, ssa.DefaultApplyOptions())
+}
+
+func waitForSet(kubeClient ctrlclient.Client, rcg genericclioptions.RESTClientGetter, changeSet *ssa.ChangeSet) error {
+	man, err := newManager(kubeClient, rcg)
+	if err != nil {
+		return err
+	}
+
+	return man.WaitForSet(changeSet.ToObjMetadataSet(), ssa.WaitOptions{Interval: 2 * time.Second, Timeout: time.Minute})
+}
+
+func Apply(kubeClient ctrlclient.Client, ctx context.Context, rcg genericclioptions.RESTClientGetter, opts *runclient.Options, manifestPath string, manifestContent []byte) (string, error) {
+	objs, err := ssa.ReadObjects(bytes.NewReader(manifestContent))
+
+	if err != nil {
+		return "", err
+	}
+
+	if len(objs) == 0 {
+		return "", fmt.Errorf("no Kubernetes objects found at: %s", manifestPath)
+	}
+
+	if err := ssa.SetNativeKindsDefaults(objs); err != nil {
+		return "", err
+	}
+
+	changeSet := ssa.NewChangeSet()
+
+	// contains only CRDs and Namespaces
+	var stageOne []*unstructured.Unstructured
+
+	// contains all objects except for CRDs and Namespaces
+	var stageTwo []*unstructured.Unstructured
+
+	for _, u := range objs {
+		if ssa.IsClusterDefinition(u) {
+			stageOne = append(stageOne, u)
+		} else {
+			stageTwo = append(stageTwo, u)
+		}
+	}
+
+	if len(stageOne) > 0 {
+		cs, err := applySet(kubeClient, ctx, rcg, stageOne)
+		if err != nil {
+			return "", err
+		}
+
+		changeSet.Append(cs.Entries)
+	}
+
+	if err := waitForSet(kubeClient, rcg, changeSet); err != nil {
+		return "", err
+	}
+
+	if len(stageTwo) > 0 {
+		cs, err := applySet(kubeClient, ctx, rcg, stageTwo)
+		if err != nil {
+			return "", err
+		}
+
+		changeSet.Append(cs.Entries)
+	}
+
+	return changeSet.String(), nil
+}
+
 func betaRunCommandRunE(opts *config.Options, client *resty.Client) func(*cobra.Command, []string) error {
 	return func(cmd *cobra.Command, args []string) error {
+		// If there is no cluster in the kube config, return an error.
+		cfg, ctx, err := kube.RestConfig()
```

Comment:
Does using `RestConfig` work equivalently/errs the same way to the `outOfClusterConfig` (@ozamosi suggested) in our case?

Or do I need to copy-paste/expose code from the `outOfClusterConfig`

How is `InClusterConfig` different from `outOfClusterConfig`, why int he `RestConfig` we first try to create the first and if it fails the second one?